### DISCUSSION TITLE
Calculating message size

### DISCUSSION THREAD

**Discussion by steve999lee** [vernemq/vernemq#1911] [Q&A]
Hi

I'm looking to restrict the message size being published by a client and to break a message which exceeds the maximum size into multiple messages. That way I can choose an reasonably small message size which handles the majority of messages but still handle the occasional larger message.

How can I calculate the size of the message (which will be checked by vernemq against max size) before I send it?

I'm using mqtt5 so there are also properties.

Thanks

**Comment by ioolkos** [comment]
Hi,
Thanks, `max_message_size` is the payload in bytes. 
If I understand correctly you want to do the breakup in smaller messages in the client (the broker can't do it anyway).
And if the clients sees that the payload is going to be too big, it'll break up the message.

I'd just test it against a given `max_message_size` value and see that you can consistently trigger it with a big message, and then consistently fly under it with multiple messages.

**Reply by steve999lee** [reply]
Thanks for your reply 

Yes that's exactly the use case - however the size being checked seems to also include properties (mqtt5) at least. i.e. if we add extra properties it can take it over the limit.
What we've done is limit payload in the publishing client to 3000 where the max_message_size is 4096 which should give us ample room for properties etc. now and if we add new ones in the future.
However this is largely guesswork and I was hoping we could calculate the message_size (including properties) before we publish it more accurately.
